indico/modules/categories/serialize.py

```diff
@@ -55,7 +58,12 @@ def serialize_category_ical(category, user, event_filter):
                       own_room_strategy,
                       own_venue_strategy)
              .order_by(Event.start_dt))
-    events = [e for e in query if e.can_access(user)]
+    if update_query:
+        query = update_query(query)
+    events = query.all()
```

Comment:
Doesn't really make a difference - all rows are retrieved either way, with `.all()` the difference is that it puts them in a list immediately, otherwise the mapping of rows to objects happens once by one while iterating.

IRC we went for it since otherwise naming would be tricky since one of the iterations is conditional and reassigning query with something not a query would be ugly. `events = query` is slightly ugly too.